_rules/audio-as-media-alternative-afb423.md

```diff
@@ -40,7 +41,7 @@ The label (from expectation 2) is [visible][] and [included in the accessibility
 
 ## Assumptions
 
-_There are currently no assumptions_
+This rule assumes that the language of the page can be correctly determined, either programmatically or by analysing the content.
```

Comment:
If I'm understanding the reason for all these changes correctly, the language stuff is being added because understanding the text relevant for the rule is necessary to determine if the success criteria is being met. 

If that is correct, then I have a couple of thoughts:

1 - Language (as in a `lang=""` attribute somewhere in the code) is not an input aspect in the same way as the page content, because understanding is more about the tester than the test subject. Though I do think the `lang="en"` attibute is a helpful addition to the examples.

2 - The assumption phrase should also emphasis understanding rather than being able to determine what the language is. Perhaps something like:
"This rule assumes that the language of the page can be correctly determined and understood, either by analysing or programmatically translating the content."

Reply:
I think there is a misunderstanding on what "input aspect" really mean. I've been through it myself many times, so no worries…

As far as I understand, "input aspect" does not mean "in order to evaluate that rule, you need a DOM tree and an accessibility tree as input". But it rather mean "in order to evaluate that rule, you need to be able to process and manipulate DOM tree and accessibility tree".

So the input aspect is not really about the test subject ("here is a DOM tree, what's the outcome?") but really about the tester ("Oh, you know DOM tree? Then what can you say about this?")

Similarly, adding language as input aspect does not mean that a language must be provided to the rule (notably, language can be inferred from the content itself) but rather than in order to test the rule, you must be able to process language to some extend.

In a crude way (assuming AI from last millennium), having the language has input aspect means "you'll need to ask a human for help to actually understand stuff here".

Reply:
That sort of makes sense ... so whatever or whoever is doing the testing will need to have/input some language understanding.

And there is an assumption that it is possible to determine that the thing being tested is correctly describing or equivalent to the thing that is being described or an alternative for, because the language could in some way be understood to make that judgement.

With that point of view, I guess Language is an Input Aspect. However, I think the assumption phrase still needs some adjustment. At the moment it could be misinterpreted to only figuring out what the language is, as opposed to understanding enough of it to determine whether or not the rule is passed.

Reply:
```suggestion
This rule assumes that the language of the page can be correctly determined and sufficiently understood, either programmatically or by analysing the content.
```

Reply:
I rather disagree.
I think this would fall in the case that @WilcoFiers was mentioning during the CG call of "unable to even provide result" rather than "risking to provide wrong result".

That is, if I can correctly determine that a page is in Chinese, I will not even try to check if the headings are descriptive. It's not that I'm going to produce false positives, I will just drop the rule and forget about it because I cannot understand the language.

Reply:
If its assumed the language can be "determined" then you have the situation of working out what the language is and trying to apply the rule. It is possible to determine a language without understanding it.

I suggested the wording "determined and sufficiently understood" because, as you say, understanding is needed to work out if something is descriptive.

Reply:
@EmmaJP Is it good now?

Reply:
Yes, that phrasing is good.